File: objective-c/HomeIndicatorHide/HomeIndicatorHide_v1/HomeIndicatorHide_v1/ViewControllers/MainViewController.m
//
//  MainViewController.m
//  HomeIndicatorHide_v1
//
//  Created by Alexandr Kirilov on 23.07.2021.
//

#import "MainViewController.h"

@interface MainViewController ()

@end

@implementation MainViewController

- (instancetype)init {
	
	self = [super init];
	if (self) {
		
		_pHideHomeIndicator = NO;
		
	}
	return self;
}

- (void)loadView {
	
	[super loadView];
	
	_pButton = [UIButton buttonWithType:UIButtonTypeCustom];
	[_pButton addTarget:self action:@selector(mSetHomeIndicatorAutoHidden) forControlEvents:UIControlEventTouchUpInside];
	[_pButton setFrame:CGRectMake(0, 0, 200, 50)];
	[_pButton setBackgroundColor:__COLOR_PURPLE];
	[_pButton setCenter:self.view.center];
	
	[self mSetButtonTitle];
	
	[self.view addSubview:_pButton];
	
}

- (void)viewDidLoad {
    
	[super viewDidLoad];
	[self.view setBackgroundColor:__COLOR_WHITE];
}

- (BOOL)prefersHomeIndicatorAutoHidden {
	
	return _pHideHomeIndicator;
}

- (void)mSetHomeIndicatorAutoHidden {
	
	if (_pHideHomeIndicator) {
		_pHideHomeIndicator = NO;
	} else {
		_pHideHomeIndicator = YES;
	}
	
	if (@available(iOS 11.0, *)) {
		[self setNeedsUpdateOfHomeIndicatorAutoHidden];
	}
	
	[self mSetButtonTitle];
}

- (void)mSetButtonTitle {
	
	if (_pHideHomeIndicator) {
		[_pButton setTitle:@"Show Home Indicator" forState:UIControlStateNormal];
	} else {
		[_pButton setTitle:@"Hide Home Indicator" forState:UIControlStateNormal];
	}
}

@end

File: swift/HomeIndicatorHide/HomeIndicatorHide_v2/HomeIndicatorHide_v2/ViewControllers/MainViewController.swift
//
//  MainViewController.swift
//  HomeIndicatorHide_v2
//
//  Created by Alexandr Kirilov on 25.07.2021.
//

import UIKit

class MainViewcontroller: UIViewController {
	
	var pHideHomeIndicator: Bool = false;
	let pButton: UIButton = UIButton();
	
	override func loadView() {
		
		super.loadView();
		self.view.backgroundColor = __COLOR_BLUE_DARK;
		
		pButton.addTarget(self, action: #selector(mActionButton), for: .touchUpInside);
		pButton.frame = CGRect(x: 0,y: 0,width: 250,height: 60);
		pButton.mSetBackgroundColor(inColor: __COLOR_PURPLE, inState: .normal);
		pButton.mSetBackgroundColor(inColor: __COLOR_WHITE, inState: .highlighted);
		pButton.setTitleColor(__COLOR_BLUE_DARK, for: .highlighted);
		pButton.center = self.view.center;
		self.mSetButtonTitle();
		self.view.addSubview(pButton);
	}
	
	override func viewDidLoad() {
		
		super.viewDidLoad();
		self.title = "Hide HomeBar Programmatically";
	}
	
	@objc func mActionButton() {
		
		if (pHideHomeIndicator) {
			pHideHomeIndicator = false;
		} else {
			pHideHomeIndicator = true;
		}
		if #available(iOS 11.0, *) {
			self.setNeedsUpdateOfHomeIndicatorAutoHidden();
		}
		self.mSetButtonTitle();
	}
	
	private func mSetButtonTitle() {
		
		if (self.pHideHomeIndicator) {
			pButton.setTitle("Show", for: .normal);
		} else {
			pButton.setTitle("Hide",for: .normal);
		}
	}
	
	override var prefersHomeIndicatorAutoHidden: Bool {
		
		return pHideHomeIndicator;
	}
}


